compiler/rustc_public/src/lib.rs

```diff
@@ -49,31 +53,25 @@ pub mod compiler_interface;
 pub mod error;
 pub mod mir;
 pub mod target;
+#[cfg(test)]
+mod tests;
 pub mod ty;
 pub mod visitor;
 
 /// Use String for now but we should replace it.
 pub type Symbol = String;
 
 /// The number that identifies a crate.
-pub type CrateNum = usize;
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct CrateNum(pub(crate) usize, ThreadLocalIndex);
```

Comment:
Sorry, I just remembered that CrateNum can be sync + send. So there's no reason to change it.